packages/notebook/src/model.ts

```diff
@@ -217,16 +221,21 @@ export class NotebookModel extends DocumentModel implements INotebookModel {
   fromJSON(value: nbformat.INotebookContent): void {
     const cells: ICellModel[] = [];
     const factory = this.contentFactory;
+    const useId = value.nbformat === 4 && value.nbformat_minor >= 5;
     for (const cell of value.cells) {
+      const options: CellModel.IOptions = { cell };
+      if (useId) {
+        options.id = (cell as any).id;
+      }
```

Comment:
Tests:
- [ ] if notebook format is 4.5, cell ids in a notebook are preserved from the input JSON to the notebook model for code, markdown, and raw cells.